.github/workflows/build-instructions.yml

```diff
@@ -116,3 +116,40 @@ jobs:
 
     - name: Run Python tests
       run: pytest src/openassetio-python/tests/package
+
```

Comment:
[fix] Capturing offline chat:

Be great to have local coverage of this. So add `ctest` checks that:
 - `python -m pip list` contains `openassetio` with the correct version (can use the cmake version)
 - no `RECORD` file exists, as this is the ["public API"](https://packaging.python.org/en/latest/specifications/recording-installed-packages/#intentionally-preventing-changes-to-installed-packages) for preventing removal.

Reply:
Ended up with 82eacd40 - using pytest and `importlib`, since testing cross-platform using shell commands got rather unwieldly.